Index: ffmpeg.svn/configure
===================================================================
--- ffmpeg.svn.orig/configure	2007-03-05 20:57:28.000000000 +0100
+++ ffmpeg.svn/configure	2007-03-05 20:57:34.000000000 +0100
@@ -1446,7 +1446,7 @@
         if test -n "`$cc -v 2>&1 | grep version | grep Apple`"; then
             add_cflags "-faltivec"
         else
-            add_cflags "-maltivec -mabi=altivec"
+            add_cflags "-maltivec"
         fi
     fi
 fi
