Index: ffmpeg.svn/configure
===================================================================
--- ffmpeg.svn.orig/configure	2007-03-05 20:57:37.000000000 +0100
+++ ffmpeg.svn/configure	2007-03-05 20:57:39.000000000 +0100
@@ -1551,6 +1551,10 @@
 # check for some common methods of building with pthread support
 # do this before the optional library checks as some of them require pthreads
 if enabled pthreads; then
+    if test "$targetos" = "Linux" -a \
+            "$cpu" = "mips" ; then
+        add_ldflags -lpthread
+    fi
     if check_func pthread_create; then
         :
     elif check_func pthread_create -pthread; then
