Index: ffmpeg-0.cvs20070307/configure
===================================================================
--- ffmpeg-0.cvs20070307.orig/configure	2007-03-09 15:42:57.000000000 +0100
+++ ffmpeg-0.cvs20070307/configure	2007-03-09 15:44:13.000000000 +0100
@@ -820,7 +820,7 @@
 VHOOKSHFLAGS='$(SHFLAGS)'
 LIBOBJFLAGS=""
 FFLDFLAGS=-Wl,--warn-common
-LDLATEFLAGS='-Wl,-rpath-link,\$(BUILD_ROOT)/libavcodec -Wl,-rpath-link,\$(BUILD_ROOT)/libavformat -Wl,-rpath-link,\$(BUILD_ROOT)/libavutil'
+LDLATEFLAGS='-L\$(BUILD_ROOT)/libavcodec -L\$(BUILD_ROOT)/libavformat -L\$(BUILD_ROOT)/libavutil -L\$(BUILD_ROOT)/libswscale -Wl,-rpath-link,\$(BUILD_ROOT)/libavcodec -Wl,-rpath-link,\$(BUILD_ROOT)/libavformat -Wl,-rpath-link,\$(BUILD_ROOT)/libavutil -Wl,-rpath-link,\$(BUILD_ROOT)/libswscale'
 FFSERVERLDFLAGS=-Wl,-E
 LDCONFIG="ldconfig"
 LIBPREF="lib"
