Index: ffmpeg-0.cvs20070307/vhook/Makefile
===================================================================
--- ffmpeg-0.cvs20070307.orig/vhook/Makefile	2007-03-09 15:15:49.000000000 +0100
+++ ffmpeg-0.cvs20070307/vhook/Makefile	2007-03-09 15:42:19.000000000 +0100
@@ -9,16 +9,21 @@
 HOOKS=null$(SLIBSUF) fish$(SLIBSUF) ppm$(SLIBSUF) watermark$(SLIBSUF)
 ALLHOOKS=$(HOOKS) imlib2$(SLIBSUF) drawtext$(SLIBSUF)
 
+LIBS_fish$(SLIBSUF) += -lavcodec -lswscale -lavformat
+LIBS_null$(SLIBSUF) += -lavcodec -lswscale
+LIBS_ppm$(SLIBSUF) += -lavcodec -lswscale -lavformat
+LIBS_watermark$(SLIBSUF) += -lavcodec -lswscale -lavformat
+
 ifeq ($(HAVE_IMLIB2),yes)
     HOOKS += imlib2$(SLIBSUF)
     CFLAGS += `imlib2-config --cflags`
-    LIBS_imlib2$(SLIBSUF) = `imlib2-config --libs`
+    LIBS_imlib2$(SLIBSUF) = `imlib2-config --libs` -lswscale -lavcodec
 endif
 
 ifeq ($(HAVE_FREETYPE2),yes)
     HOOKS += drawtext$(SLIBSUF)
     CFLAGS += `freetype-config --cflags`
-    LIBS_drawtext$(SLIBSUF) = `freetype-config --libs`
+    LIBS_drawtext$(SLIBSUF) = `freetype-config --libs` -lavutil
 endif
 
 SRCS := $(HOOKS:$(SLIBSUF)=.c)
