Index: ffmpeg-0.cvs20070307/libavcodec/Makefile
===================================================================
--- ffmpeg-0.cvs20070307.orig/libavcodec/Makefile	2007-03-07 14:37:04.000000000 +0100
+++ ffmpeg-0.cvs20070307/libavcodec/Makefile	2007-03-28 16:54:03.000000000 +0200
@@ -310,9 +310,7 @@
 
 OBJS-$(HAVE_XVMC_ACCEL)                += xvmcvideo.o
 
-ifneq ($(CONFIG_SWSCALER),yes)
 OBJS += imgresample.o
-endif
 
 # i386 mmx specific stuff
 ifeq ($(TARGET_MMX),yes)
Index: ffmpeg-0.cvs20070307/tests/Makefile
===================================================================
--- ffmpeg-0.cvs20070307.orig/tests/Makefile	2007-03-07 14:37:04.000000000 +0100
+++ ffmpeg-0.cvs20070307/tests/Makefile	2007-03-28 16:18:43.000000000 +0200
@@ -43,14 +43,6 @@
 	@echo
 endif
 
-ifeq ($(CONFIG_SWSCALER),yes)
-test-server codectest mpeg4 mpeg ac3 snow snowll libavtest: swscale_error
-swscale_error:
-	@echo
-	@echo "This regression test is incompatible with --enable-swscaler."
-	@echo
-	@exit 1
-endif
 
 seektest: seek_test$(EXESUF)
 	@$(SRC_DIR)/seek_test.sh $(SEEK_REFFILE)
Index: ffmpeg-0.cvs20070307/libavcodec/imgconvert.c
===================================================================
--- ffmpeg-0.cvs20070307.orig/libavcodec/imgconvert.c	2007-03-28 16:54:10.000000000 +0200
+++ ffmpeg-0.cvs20070307/libavcodec/imgconvert.c	2007-03-28 16:54:20.000000000 +0200
@@ -2317,7 +2317,7 @@
 }
 #endif
 
-#ifndef CONFIG_SWSCALER
+#if 1
 /* XXX: always use linesize. Return -1 if not supported */
 int img_convert(AVPicture *dst, int dst_pix_fmt,
                 const AVPicture *src, int src_pix_fmt,
