Index: ffmpeg-0.cvs20060823/libavformat/rm.c
===================================================================
--- ffmpeg-0.cvs20060823.orig/libavformat/rm.c	2007-03-23 22:42:34.000000000 +0100
+++ ffmpeg-0.cvs20060823/libavformat/rm.c	2007-03-23 22:43:28.000000000 +0100
@@ -959,9 +959,8 @@
                 len=len2;
             rm->remaining_len-= len;
             av_get_packet(pb, pkt, len);
-        }
 
-        if (st->codec->codec_type == CODEC_TYPE_AUDIO) {
+        } else if (st->codec->codec_type == CODEC_TYPE_AUDIO) {
             if ((st->codec->codec_id == CODEC_ID_RA_288) ||
                 (st->codec->codec_id == CODEC_ID_COOK)) {
                 int x;
@@ -1013,7 +1012,8 @@
                 }
             } else
                 av_get_packet(pb, pkt, len);
-        }
+
+        } else goto resync;
 
         if(  (st->discard >= AVDISCARD_NONKEY && !(flags&2))
            || st->discard >= AVDISCARD_ALL){
