--- ffmpeg-0.cvs20070307.orig/libavcodec/avcodec.h	2007-03-07 14:37:04.000000000 +0100
+++ ffmpeg-0.cvs20070307/libavcodec/avcodec.h	2007-08-06 20:40:11.000000000 +0200
@@ -2421,7 +2421,7 @@
 #if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 /* YUV420 format is assumed ! */
 
-struct ImgReSampleContext attribute_deprecated;
+struct ImgReSampleContext;
 
 typedef struct ImgReSampleContext ImgReSampleContext attribute_deprecated;
 
@@ -2436,10 +2436,10 @@
                                       int padleft, int padright);
 
 
-attribute_deprecated void img_resample(ImgReSampleContext *s,
+attribute_deprecated void img_resample(struct ImgReSampleContext *s,
                   AVPicture *output, const AVPicture *input);
 
-attribute_deprecated void img_resample_close(ImgReSampleContext *s);
+attribute_deprecated void img_resample_close(struct ImgReSampleContext *s);
 
 #endif
 
